onnxruntime/core/graph/graph.cc

```diff
@@ -2353,15 +2331,28 @@ Status Graph::SetGraphInputsOutputs() {
           continue;
         }
 
-        auto output_arg_iter = output_name_to_node_arg.find(input_arg->Name());
-        if (output_name_to_node_arg.end() == output_arg_iter) {
+        auto output_arg_iter = output_name_to_node_arg_index.find(input_arg->Name());
+        if (output_name_to_node_arg_index.end() == output_arg_iter) {
           // This input arg should be fed when running evaluation.
           // it should be a graph input.
           const std::string& name = input_arg->Name();
           if (added_input_names.end() == added_input_names.find(name)) {
             // This graph input has not been added into <graph_inputs_>.
-            graph_inputs_including_initializers_.push_back(input_arg);
-
+            if (!graph_inputs_manually_set) {
+              graph_inputs_including_initializers_.push_back(input_arg);
+            } else {
+              // Validation: the <input_arg> must be in graph inputs or initializers when it's manually set.
+              auto& inputs = GetInputsIncludingInitializers();
+              auto iter = std::find(inputs.begin(), inputs.end(), input_arg);
+              if (inputs.end() == iter) {
+                // it's not in graph inputs.
+                auto initializers = GetAllInitializedTensors();
+                if (initializers.end() == initializers.find(input_arg->Name())) {
```

Comment:
Any reason to call a method that returns an internal of the same class, of can you just use graph_inputs_including_initializers_ and name_to_initial_tensor_ directly?

Reply:
I suggest to use the method if it's exposed by a public API. Any preferences or guidelines of not doing that?

Reply:
In the case of an accessor that just returns a member variable, it seems unnecessary to use the public API. I think it also makes it harder when searching the implementation of a class to find usages of a member if that is via the API in places as now you have to search for two different things.